merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/generator/MissionModelGenerator.java

```diff
@@ -646,6 +586,89 @@ public JavaFile generateActivityTypes(final MissionModelRecord missionModel) {
         .build();
   }
 
+  /** Generate `ActivityTypes.Builder` class. */
+  private static TypeSpec generateActivityTypesBuilder(final MissionModelRecord missionModel) {
+    final var activityFields = generateActivityDirectiveFields(missionModel, Modifier.PRIVATE);
+    return TypeSpec.classBuilder("Builder")
+        .addModifiers(Modifier.PRIVATE, Modifier.STATIC, Modifier.FINAL)
+        .addFields(activityFields)
+        .addMethods(activityFields.stream()
+            .map(af ->
+                MethodSpec.methodBuilder(af.name)
+                    .addModifiers(Modifier.PRIVATE)
+                    .returns(missionModel.getTypesName().nestedClass("Builder"))
+                    .addParameter(af.type, af.name, Modifier.FINAL)
+                    .addStatement(
+                        "this.$L = $L",
+                        af.name,
+                        af.name)
+                    .addStatement("return this")
+                    .build())
+            .toList())
+        .build();
+  }
+
+  /** Generate `ActivityTypes.Builder` class. */
+  private static MethodSpec generateActivityTypesRegisterMethod(final MissionModelRecord missionModel) {
+    final var typeName = missionModel.getTypesName();
+
+    final var cb = CodeBlock.builder()
+        .add(
+            "return new $T(new $T()",
```

Comment:
It's a little hard for me to follow this string template across the method. In particular, the matching right parenthesis is much further down, and it's not clear how the expression gets continued. Are the rest of the `cb.add` calls adding parameters, or chaining method calls? I'm not sure either of those is what's happening, but I can't tell. (It looks like `$W` is just a line-wrapping directive, so we're building up fluent method calls -- both of these facts could be made immediately clear with a comment.)

Personally, I'd like the wrapping `return new $T(` and `)` to be moved into the `addStatement` in the `MethodSpec` at the end of this method e.g.

```java
.addStatement("return new $T($L)", cb.build())
```

Then `cb` (which should be named something like `builderSteps`) can be dedicated solely to building up the sequence of steps on the registrar.

Reply:
Agreed, thanks for jotting this down. I'm cool with your suggestion! Small PR coming out.

Reply:
#286 :)